File: SunnyWeather-pro/src/main/java/com/example/sunnyweather_pro/logic/network/ServiceCreator.java
package com.example.sunnyweather_pro.logic.network;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

//创建一个Retrofit构建器
public class ServiceCreator {

    //彩云天气网络根地址
    private static final String BASE_URL = "https://api.caiyunapp.com/";

    //retrofit实例
    private static final Retrofit retrofit = new Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build();

    //返回Service接口对象
    public static <T> T create(Class<T> serviceClass){
        return retrofit.create(serviceClass);
    }

}

File: SunnyWeather-kotlin/src/main/java/com/driverskr/sunnyweather_kotlin/logic/network/ServiceCreator.kt
package com.driverskr.sunnyweather_kotlin.logic.network

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

/**
 * @Author: driverSkr
 * @Time: 2023/11/6 15:13
 * @Description: Service生成器$
 */
object ServiceCreator {

    private const val BASE_URL = "https://api.caiyunapp.com/"

    private val retrofit = Retrofit.Builder()
        .baseUrl(BASE_URL)
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    fun <T> create(serviceClass: Class<T>): T = retrofit.create(serviceClass)

    inline fun <reified T> create(): T = create(T::class.java)
}